### PR TITLE
CDC-ACM USB serial fixes

### PR DESCRIPTION
**Pull request by bigdinotech**
-improve reliability of USB serial functionality
-increase stack size for threads used by USB serial services

### CODE REVIEW

**Comment by bigdinotech** [comment]
@eriknyquist @SidLeung ready for review

File: src/cdcacm_serial.c [hunk 1]
```diff
@@ -89,9 +89,7 @@ static void write_data(struct device *dev, const char *buf, int len)
 	uart_fifo_fill(dev, (const uint8_t*)buf, len);
 	while (data_transmitted == false)
```
Comment (eriknyquist) [inline src/cdcacm_serial.c]: what are the implications of removing this yield() call? does it mean that x86 will be stuck busy waiting in this thread until the whole string is written?

File: src/cdcacm_serial.c [hunk 2]
```diff
@@ -101,14 +99,13 @@ void cdc_acm_tx()
 	{
 		if(Tx_HEAD != Tx_TAIL)
 		{
-			int cnt = 0, index = Tx_TAIL;
+			int cnt = 0;
 			gpio_pin_write(gpio_dev, TXRX_LED, 0);	//turn TXRX led on
-			for (; (index != Tx_HEAD) && (cnt < BUFFER_LENGTH);cnt++) 
+			for (; (Tx_TAIL != Tx_HEAD) && (cnt < BUFFER_LENGTH*2);cnt++)
```
Comment (eriknyquist) [inline src/cdcacm_serial.c]: weird.... how come you're doing the initialization on line 102 instead of in the for loop?

Comment (eriknyquist) [inline src/cdcacm_serial.c]: meh I guess it was already there...... still unusual for no good reason

File: src/cdcacm_serial.c (hunk 1 above)
Comment (bigdinotech) [inline src/cdcacm_serial.c]: I actually noticed that having the yield it gets stuck inside the while loop. 
The originall cdc-acm example from Zephyr does not have the yield in there.

**Pull request merged**